### PR TITLE
[storage-blob] renames LeaseClient to BlobLeaseClient

### LINKED ISSUE #8110: Rename LeaseClient in blobs to BlobLeaseClient

### PR DESCRIPTION
**Pull request by kristapratico**
Resolves #8110

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_upload_helpers.py [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 if TYPE_CHECKING:
     from datetime import datetime # pylint: disable=unused-import
-    LeaseClient = TypeVar("LeaseClient")
+    BlobLeaseClient = TypeVar("BlobLeaseClient")
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_upload_helpers.py]: Was this var unused in the rest of the file?
If it is, don't worry about it, probably just indicates we need a more thorough mypy pass - but just checking that no other instances needed renaming.

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py [hunk 2]
```diff
@@ -948,8 +948,8 @@ def __init__(
             The match condition to use upon the etag.
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
-        :returns: A LeaseClient object.
-        :rtype: ~azure.storage.blob.aio.lease_async.LeaseClient
```
Comment (annatisch) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_client_async.py]: Awesome - you caught lots of docs cross-ref issues as well :)

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_upload_helpers.py (hunk 1 above)
Comment (kristapratico) [reply sdk/storage/azure-storage-blob/azure/storage/blob/aio/_upload_helpers.py]: Looks like it is unused. I can remove it when I make my docstring PR

**Pull request merged**